cmake/ConfigureCompiler.cmake

```diff
@@ -346,9 +346,19 @@ else()
     set(DTRACE_PROBES 1)
   endif()
 
-  if(CMAKE_COMPILER_IS_GNUCXX)
-    set(USE_LTO OFF CACHE BOOL "Do link time optimization")
-    if (USE_LTO)
+  set(USE_LTO OFF CACHE BOOL "Do link time optimization")
+  if (USE_LTO)
+    if (CLANG)
+      if (LLVM_ENABLE_LTO STREQUAL "Full")
+        add_compile_options($<$<CONFIG:Release>:-flto=full>)
+      else()
+        add_compile_options($<$<CONFIG:Release>:-flto=thin>)
+        set(LLVM_ENABLE_LTO "Thin")
+      endif()
+      set(CMAKE_RANLIB "/usr/local/bin/llvm-ranlib")
```

Comment:
Shouldn't we use relative paths?
Although the official `foundationdb/build` image has those binaries under `/usr/local/bin`, we still should support any environment.

Reply:
done